zerver/worker/queue_processors.py

```diff
@@ -613,27 +614,53 @@ class MissedMessageWorker(QueueProcessingWorker):
     stopping = False
     has_timeout = False
 
+    cursor: Optional[CursorWrapper] = None
+
     # The main thread, which handles the RabbitMQ connection and creates
     # database rows from them.
     @override
     def consume(self, event: Dict[str, Any]) -> None:
+        if self.cursor is None:
```

Comment:
Will this have problems recovering if the database server restarts? We might need something to clear `self.cursor` in the event of whatever exception that throws.